lib/saml/user_attributes/mhv.rb

```diff
@@ -2,36 +2,76 @@
 require 'saml/user_attributes/base_decorator'
 
 # TODO: remove these nocov comments when this is able to be tested.
-#:nocov:
+# :nocov:
 module SAML
   module UserAttributes
     class MHV < BaseDecorator
       def mhv_icn
-        attributes['mhv_icn']&.first
-      end
-
-      def mhv_profile
-        JSON.parse(attributes['mhv_profile']&.first)
+        attributes['mhv_icn']
       end
 
+      # NOTE: this is derived from mhv_profile which is a complex JSON object, see above
       def account_type
         mhv_profile['accountType']
       end
 
+      # NOTE: this is derived from mhv_profile which is a complex JSON object, see above
+      # QUESTION: Why is this premium user only listed to have Blue Button and not Rx or SM?
       def available_services
         mhv_profile['availableServices']
       end
 
-      def mhv_uuid
+      # NOTE: this is the same thing as mhv_correlation_id it should supercede any mhv
+      # mhv correlation id that is returned by MVI
+      def mhv_correlation_id
         attributes['mhv_uuid']
       end
 
+      def uuid
+        attributes['uuid']
+      end
+
+      # NOTE: This attribute is originated by id.me
+      def email
+        attributes['email']
+      end
+
+      # NOTE: This attribute is originated by id.me
+      def multifactor
+        attributes['multifactor']
+      end
+
+      # NOTE: See comments for loa_current and loa_highest below
+      def loa
+        { current: loa_current, highest: loa_highest }
+      end
+
       private
 
+      # NOTE: email, uuid, loa are derived values, all others originate from MHV
+      # For now we will probably not use available services, mhv profile is unnecessary
+      # once we have the base components
       def serializable_attributes
-        %i(mhv_icn mhv_profile account_type available_services mhv_uuid)
+        %i(mhv_icn email uuid loa multifactor)
+      end
+
+      # if the account type is premium then the user has identity proofed with MHV and we trust it
+      def loa_current
+        account_type == 'Premium' ? 2 : 1
+      end
+
+      # if the account type is premium there is no option to FICAM level up the account, so the highest is
+      # the current level of 2. If however the user is Basic or Advanced, they should have the option
+      # to level up their account using ID.me similar to other ID.me login users
+      def loa_highest
```

Comment:
As discussed on Slack, this implements very different semantics for loa_highest between this class and the idme user attributes class.

Reply:
I think this is ok. I'd like to hear arguments against this, perhaps the concept needs to change entirely in light of the fact that our concept of LOA is a little bit more complicated now. But because MFA is now separated, and LOA3 is ONLY FICAM LOA3, I think it makes sense for us to revise our meaning of loa_highest.

Reply:
I changed the logic on loa_highest. Additionally changed logic on loa_current to = 3 if ds_logon = "premium", this will have the least impact to the F/E and how we gate authorization for now. 

It's the least complexity. We'll have to look at authorization more holistically later.